### PR TITLE
feat: implement DELEGATECALL and CALLCODE

### PR DESCRIPTION
**Pull request by daejunpark**

### CODE REVIEW

File: tests/test/TestConstructor.t.sol [hunk 1]
```diff
@@ -6,16 +6,26 @@ contract TestConstructorTest {
...
+    uint codesize_;
+    uint extcodesize_;
 
     constructor () {
         flag = 3;
         value = 4;
+
+        assembly {
+            sstore(codesize_.slot, codesize())
+            sstore(extcodesize_.slot, extcodesize(address()))
```
Comment (karmacoma-eth) [inline tests/test/TestConstructor.t.sol:18]: ah interesting 👌

File: tests/test/Call.t.sol [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+        assert(address(this).balance == 0);
+        assert(address(d).balance == fund);
+        assert(address(d.c()).balance == 0);
+    }
+
+    function check_delegatecall(uint x, uint fund) public payable {
+        vm.deal(address(this), fund);
+        vm.deal(address(d), 0);
+        vm.deal(address(d.c()), 0);
```
Comment (karmacoma-eth) [inline tests/test/Call.t.sol:107]: why the deal 0s?

**Review by karmacoma-eth — APPROVED** [review]
looks great!

File: tests/test/Call.t.sol (hunk 2 above)
Comment (daejunpark) [reply tests/test/Call.t.sol:107]: balance is not assumed to be zero, because someone might have already sent fund to accounts before created.

**Pull request merged**